qcow/src/qcow.rs

```diff
@@ -258,7 +258,7 @@ impl QcowHeader {
         let l2_size: u32 = cluster_size / size_of::<u64>() as u32;
         let num_clusters: u32 = div_round_up_u64(size, u64::from(cluster_size)) as u32;
         let num_l2_clusters: u32 = div_round_up_u32(num_clusters, l2_size);
-        let l1_clusters: u32 = div_round_up_u32(num_l2_clusters, cluster_size);
+        let l1_clusters: u32 = div_round_up_u32(num_l2_clusters, l2_size);
```

Comment:
- `l2_size` should really be called `entries_per_cluster` given we're dividing a cluster size by size_of(u64), which really means we want the number of entries/addresses that can fit into a cluster.
- `num_clusters` looks good given we're dividing the total size in bytes by the size of a cluster in bytes.
- `num_l2_clusters` is completely wrong. It's dividing the total number of clusters by the number of entries in a cluster, which are not the same unit. `num_l2_clusters` should be the same as `num_clusters` given a l2 table is one cluster in size.
- `l1_clusters` which should be called `num_l1_clusters` is also wrong (well you actually fixed it but it requires `num_l2_clusters` to be correct).

So here is my suggestion to make things a bit clearer:
```rust
let entries_per_cluster: u32 = cluster_size / size_of::<u64>() as u32;
let num_clusters: u32 = div_round_up_u64(size, u64::from(cluster_size)) as u32;
let num_l2_clusters: u32 = num_clusters;
let num_l1_clusters: u32 = div_round_up_u32(num_l2_clusters, entries_per_cluster);
```